tests/ToolTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Tightenco\NovaGoogleAnalytics\Tests;
+
+use Tightenco\NovaGoogleAnalytics\Http\Controllers\GoogleAnalyticsController;
+use Spatie\Analytics\Period;
+
+class ToolTest extends TestCase
+{
+    /** @test */
+    public function it_can_return_a_response()
+    {
+        $this
+            ->get('nova-vendor/tightenco/nova-google-analytics/pages')
+            ->assertSuccessful()
+            ->assertJsonStructure([
+                'hasMore',
+                'pages',
+                'totalPages'
+            ]);
+    }
+
+    /** @test */
+    public function it_will_accept_limit_param()
+    {
+        $this
+            ->get('nova-vendor/tightenco/nova-google-analytics/pages?limit=5')
+            ->assertJsonCount(5, 'pages');
+    }
+
+    /** @test */
+    public function it_will_not_accept_non_numeric_limit()
+    {
+        $this->get('nova-vendor/tightenco/nova-google-analytics/pages?limit=a')
+            ->assertStatus(500);
+    }
+
+    /** @test */
+    public function it_will_accept_page_param()
+    {
+        $page1 = $this->get('nova-vendor/tightenco/nova-google-analytics/pages?page=1');
+        $page2 = $this->get('nova-vendor/tightenco/nova-google-analytics/pages?page=2');
+
+        $this->assertNotEquals($page1, $page2);
+    }
+
+    /** @test */
+    public function it_will_not_accept_non_numeric_page()
+    {
+        $this->get('nova-vendor/tightenco/nova-google-analytics/pages?page=a')
+            ->assertStatus(500);
+    }
+
+    /** @test */
+    public function it_will_accept_duration_param()
+    {
+        $this->get('nova-vendor/tightenco/nova-google-analytics/pages?duration=month')
+            ->assertSuccessful();
+    }
+
+    /** @test */
+    public function invalid_duration_will_default_to_week()
+    {
+        $controller = new GoogleAnalyticsController();
+        $reflection = new \ReflectionClass($controller);
+        $method = $reflection->getMethod('periodForDuration');
+        $method->setAccessible(true);
+        $duration = $method->invokeArgs($controller, ['asdf']);
+        $this->assertEquals(Period::days(7), $duration);
+    }
+
+    /** @test */
+    public function it_will_accept_search_param()
+    {
+        $this->get('nova-vendor/tightenco/nova-google-analytics/pages?s=blog')
+            ->assertSuccessful();
+    }
+
+    /** @test */
+    public function it_will_accept_invalid_search_param()
+    {
+        $this->get('nova-vendor/tightenco/nova-google-analytics/pages?s=123')
+            ->assertSuccessful();
+    }
```

Comment:
If it accepts anything, is it really invalid? I think you can just drop this one.